quarkus-logging-sentry and quarkus-undertow -  java.lang.ClassCastException: io.sentry.servlet.SentryServletContainerInitializer cannot be cast to javax.servlet.ServletContainerInitializer

When using quarkus-logging-sentry and quarkus-undertow an app user gets `java.lang.ClassCastException: io.sentry.servlet.SentryServletContainerInitializer cannot be cast to javax.servlet.ServletContainerInitializer`

```
[INFO] Running io.quarkus.qe.MyResourceTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.998 s <<< FAILURE! - in io.quarkus.qe.MyResourceTest
[ERROR] testHelloEndpoint  Time elapsed: 0.013 s  <<< ERROR!
java.lang.RuntimeException: java.lang.ExceptionInInitializerError
Caused by: java.lang.ExceptionInInitializerError
Caused by: java.lang.RuntimeException: Failed to start quarkus
Caused by: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.ClassCastException: io.sentry.servlet.SentryServletContainerInitializer cannot be cast to javax.servlet.ServletContainerInitializer
Caused by: java.lang.RuntimeException: java.lang.ClassCastException: io.sentry.servlet.SentryServletContainerInitializer cannot be cast to javax.servlet.ServletContainerInitializer
Caused by: java.lang.ClassCastException: io.sentry.servlet.SentryServletContainerInitializer cannot be cast to javax.servlet.ServletContainerInitializer
```

Reproducer:
```
rm -rf fooBar
mvn io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:create \
    -DprojectGroupId=io.quarkus.qe \
    -DprojectArtifactId=fooBar \
    -DprojectVersion=1.0.0-SNAPSHOT \
    -DplatformArtifactId=quarkus-bom \
    -DclassName="io.quarkus.qe.MyResource" \
    -Dextensions="quarkus-logging-sentry,quarkus-undertow"
mvn -f fooBar/pom.xml clean verify
```

--- a/core/runtime/pom.xml
+++ b/core/runtime/pom.xml
@@ -109,6 +109,7 @@
                         <parentFirstArtifact>org.jboss.logmanager:jboss-logmanager-embedded</parentFirstArtifact>
                         <parentFirstArtifact>org.jboss.logging:jboss-logging</parentFirstArtifact>
                         <parentFirstArtifact>org.ow2.asm:asm</parentFirstArtifact>
+                        <parentFirstArtifact>org.apache.maven:maven-model</parentFirstArtifact>
                     </parentFirstArtifacts>
                     <excludedArtifacts>
                         <excludedArtifact>io.smallrye:smallrye-config</excludedArtifact>

--- a/core/runtime/src/main/java/io/quarkus/runtime/logging/LoggingSetupRecorder.java
+++ b/core/runtime/src/main/java/io/quarkus/runtime/logging/LoggingSetupRecorder.java
@@ -121,7 +121,7 @@ public class LoggingSetupRecorder {
             final Logger categoryLogger = logContext.getLogger(name);
             final CategoryConfig categoryConfig = entry.getValue();
             final String levelName = categoryConfig.level.toUpperCase(Locale.ROOT);
-            if (!"INHERIT".equals(categoryConfig.level)) {
+            if (!"INHERIT".equals(levelName)) {
                 categoryLogger.setLevelName(levelName);
             }
             categoryLogger.setUseParentHandlers(categoryConfig.useParentHandlers);

--- a/extensions/undertow/runtime/pom.xml
+++ b/extensions/undertow/runtime/pom.xml
@@ -93,6 +93,15 @@
                         <excludedArtifact>javax.servlet:servlet-api</excludedArtifact>
                         <excludedArtifact>javax.servlet:javax.servlet-api</excludedArtifact>
                     </excludedArtifacts>
+                    <parentFirstArtifacts>
+                        <!--
+                        Sentry logging has a dependency on the Servlet API, so we load it parent first to
+                        prevent class loading problems.
+
+                        see https://github.com/quarkusio/quarkus/issues/7407
+                        -->
+                        <parentFirstArtifact>jakarta.servlet:jakarta.servlet-api</parentFirstArtifact>
+                    </parentFirstArtifacts>
                 </configuration>
             </plugin>
             <plugin>

